### PR TITLE
Replaced set-uint-at with replace-at

### PR DESCRIPTION
**Pull request by a3slade**
### Description
Replaces set-uint-at with replace-at in the pox-2 contract. Removes the now unused set-uint-at function.

### Applicable issues
- fixes https://github.com/stacks-network/stacks-blockchain/issues/3359

### CODE REVIEW

File: src/chainstate/stacks/boot/pox-2.clar
Comment (pavitthrap) [inline src/chainstate/stacks/boot/pox-2.clar:306]: you can remove this comment - or just update the latter half of it

**Review by pavitthrap — APPROVED** [review]
lgtm

**Review by jcnelson — APPROVED** [review]
LGTM.  Let's just wait for CI to finish running before merging :+1:

**Pull request merged**
